doc/configurable_mount_options.xml

```diff
@@ -0,0 +1,318 @@
+    </para>
+
+    <para>
+    Options that are not allowed in the computed <literal>_allow</literal> set will fail the mount operation with an error.
+    </para>
+  </simplesect>
+
+  <simplesect><title>Unprivileged mounts, UID and GID passing</title>
+    <para>
+    UDisks daemon typically runs as root and spawns the <command>mount</command> command with root privileges while processing requests from unprivileged clients and it needs a way to set proper permissions for the mounted filesystem.
+    This is usually done by passing the <literal>uid=</literal> and <literal>gid=</literal> mount options that common filesystems honour. UDisks take the D-Bus method caller effective UID and finds its corresponding primary GID.
+    </para>
+
+    <para>
+    Also a <literal>uhelper=udisks2</literal> mount option is added automatically to indicate the <command>umount</command> command to use UDisks again for unmounting when called by an unprivileged user.
+    </para>
+
+    <para>
+    To allow flexibility in mount option naming, two special arguments are defined as placeholders for aforementioned IDs: <literal>$UID</literal> and <literal>$GID</literal>.
+    These are supposed to be defined within the <literal>_allow</literal> set and are then replaced with numerical values in the late phase of mount option computation.
+    </para>
+
+    <para>
+    It's worth noting that there are few more options whose arguments are being replaced by computed values: <literal>mode</literal> and <literal>dmode</literal>.
+    These options are hardcoded at the moment and its values depend on the udev <literal>UDISKS_FILESYSTEM_SHARED</literal> attribute presence on the block device.
+    </para>
+  </simplesect>
+
+  <simplesect><title>Option set specifics</title>
+    <para>
+    Option sets are composite strings of mount options with or without arguments separated by commas. There's generally no difference from an option string commonly used with the <command>mount</command> command. However UDisks brings a couple of specifics for greater flexibility.
+    </para>
+
+    <para>
+    As explained above any particular mount option either defined in the <literal>_defaults</literal> set or passed as an extra argument via the <link linkend="gdbus-method-org-freedesktop-UDisks2-Filesystem.Mount">Mount()</link> method call need to be allowed by the <literal>_allow</literal> set of mount options.
+    There's however a difference in how the option is allowed and what arguments are valid.
+    </para>
+
+    <para>
+    When a considered option is defined with either of the <literal>$UID</literal> or <literal>$GID</literal> arguments within the <literal>_allow</literal> set (see previous chapter), special rules apply.
```

Comment:
I think a detailed example for the `uid=` option could be useful here. I took me a while to decipher what the combinations of `_allow=uid=` and `_defaults=uid=` do. Few examples just with these two with explanation of the result would be really useful.

Reply:
Added an example for specific UIDs at the end of the document. Also clarified the UID computation a little bit. Please check whether it's more clear now. Or even send a PR with suggested changes ;-)